### ISSUE TITLE
Plugin acts incorrect then enum variant has same name as enum

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#9522]
## Environment

* **IntelliJ Rust plugin version:** 0.4.181.4937-223-nightly
* **Rust toolchain version:** 1.64.0 (a55dd71d5 2022-09-19) x86_64-apple-darwin
* **IDE name and version:** CLion 2022.3 EAP (CL-223.6160.115)
* **Operating system:** macOS 12.5.1
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.macros.proc

## Problem description
Plugin adds import `use crate::Cat::Cat;` that breaks code, and does not detect any problems:
<img width="581" alt="image" src="https://user-images.githubusercontent.com/49211026/195593211-7064da74-f1db-4cb2-a1fa-58031a4282de.png">
Usually, the plugin detects  E0255:
<img width="646" alt="image" src="https://user-images.githubusercontent.com/49211026/195592414-d78874aa-e443-4e50-9d0c-1b7c4b5db4b9.png">

And if we add some other import from Cat enum, the plugin will detect false problem:
<img width="642" alt="image" src="https://user-images.githubusercontent.com/49211026/195591387-37e1688a-0fa4-410f-ba5f-09a471c58501.png">

## Steps to reproduce
```rust
enum Cat {
    Felix,
    Vasiliy,
    Cat,
}

fn main() {
    let my_cat = Felix; // import will works fine
    let my_second_cat = Cat; // import will break code

}

```
